src/attributes.cpp

```diff
@@ -2196,7 +2197,7 @@ namespace attributes {
     }
 
     std::string CppPackageIncludeGenerator::getHeaderGuard() const {
-        return "__" + packageCpp() + "_h__";
+        return "RCPP_" + packageCpp() + "_h_gen_";
```

Comment:
@eddelbuettel had suggested capitals (which is generally what I've seen in header guards as well). I don't have a super strong preference but I think we should be consistent, e.g. either:

```cpp
"_rcpp_" + packageCpp() + "_h_gen_"
```

or

```cpp
"_RCPP_" + packageCpp() + "_H_GEN_"
```